imports/plugins/core/i18n/client/containers/currencyDropdown.js

```diff
@@ -16,6 +16,7 @@ const handlers = {
     // UserProfile and ShopMembers publications.
     //
     Accounts.update(Meteor.userId(), { $set: { "profile.currency": currencyName } });
+    // Hooks.Events.run("afterAccountUpdate", Meteor.userId(), Meteor.user());
```

Comment:
The correct fix here is to create a new server method in `server/methods/accounts/accounts.js` that does `Accounts.update` followed by running the hook. It could be called something like `"accounts/setProfileCurrency"`. It does not need any security on it other than making sure `this.userId` is set (caller is logged in) and then using `this.userId` for the `Accounts.update` ID.